src/ui/components.tsx

```diff
@@ -68,22 +71,25 @@ export const CheckMkAsyncSelect = function <Key extends keyof RequestSpec, Value
   }
 
   const loadOptions = React.useCallback(
-    (inputValue: string): Promise<Array<SelectableValue<Value>>> => {
-      return autocompleter(inputValue).then(
-        (data) => {
-          setAutocompleteError('');
-          setOptions(data);
-          return data;
-        },
-        (error) => {
-          if (error && error.message) {
-            setAutocompleteError(error.message);
-          }
-          throw error;
+    async (inputValue: string): Promise<Array<SelectableValue<Value>>> => {
+      try {
+        const data = await autocompleter(inputValue);
+        setAutocompleteError('');
+        if (isNull(findValueInOptions(data))) {
+          const specificData = await autocompleter(value as string);
```

Comment:
the `as string` looks dangerous to me. maybe we can modify the signature of the whole React-Component, so the typing makes sure that only keys can be used which point to string data?